ts/defs.bzl

```diff
@@ -318,19 +325,43 @@ def ts_project(
     typings_out_dir = declaration_dir if declaration_dir else out_dir
     tsbuildinfo_path = ts_build_info_file if ts_build_info_file else name + ".tsbuildinfo"
 
-    tsc_typings_outs = _lib.calculate_typings_outs(srcs, typings_out_dir, root_dir, declaration, composite, allow_js, no_emit)
-    tsc_typing_maps_outs = _lib.calculate_typing_maps_outs(srcs, typings_out_dir, root_dir, declaration_map, allow_js, no_emit)
+    # Target names for tsc, dts+js transpilers
+    tsc_target_name = name
+    declarations_target_name = None
+    transpile_target_name = None
+
+    # typing outputs
+    tsc_typings_outs = []
+    tsc_typing_maps_outs = []
+    if no_emit or not declaration_transpiler:
+        tsc_typings_outs = _lib.calculate_typings_outs(srcs, typings_out_dir, root_dir, declaration, composite, allow_js, no_emit)
+        tsc_typing_maps_outs = _lib.calculate_typing_maps_outs(srcs, typings_out_dir, root_dir, declaration_map, allow_js, no_emit)
+    elif not no_emit and declaration:
```

Comment:
Note that we declare no `declarations_target_name` if we don't need the `declaration_transpiler` to output any dts files.

Reply:
I think `elif declaration:` should be sufficient since if `no_emit` is True you'll fall into the first condition

Reply:
Actually, maybe this should be
```
elif declaration or emit_declaration_only:
```
if it is valid to have `declaration` False and `emit_declaration_only` True?

Reply:
> if it is valid to have declaration False and emit_declaration_only True?

tsc fails in that case
```
error TS5069: Option 'emitDeclarationOnly' cannot be specified without specifying option 'declaration' or option 'composite'.
```
... then if trying to test that `composite` scenario:
```
examples/transpiler/tsconfig_custom_transpilers-emit_declaration_only.json(1,54): error TS6304: Composite projects may not disable declaration emit.
```
... because we are using `tsc --noEmit` due to the `declaration_transpiler` 🤯